engine-flink/src/main/java/org/apache/kylin/engine/flink/FlinkExecutable.java

```diff
@@ -205,7 +205,7 @@ protected ExecuteResult doWork(ExecutableContext context) throws ExecuteExceptio
                     //flink on yarn specific option (pattern : -yn 1)
                     if (configOptionKey.startsWith("-y") && !entry.getValue().isEmpty()) {
                         sb.append(" ").append(configOptionKey).append(" ").append(entry.getValue());
-                    } else {
+                    } else if(!configOptionKey.startsWith("-y")){
```

Comment:
Good catch! I remembered there is a bug about the config option. Thanks for your fix!